spinta/manifests/internal_sql/helpers.py

```diff
@@ -0,0 +1,1085 @@
+
+import sqlalchemy as sa
+from sqlalchemy.sql.elements import Null
+
+from spinta.backends import Backend
+from spinta.backends.components import BackendOrigin
+from spinta.components import Namespace, Base, Model, Property
+from spinta.core.enums import Access
+from spinta.core.ufuncs import Expr
+from spinta.datasets.components import Dataset, Resource
+from spinta.dimensions.comments.components import Comment
+from spinta.dimensions.enum.components import Enums
+from spinta.dimensions.lang.components import LangData
+from spinta.dimensions.prefix.components import UriPrefix
+from spinta.manifests.components import Manifest
+from spinta.manifests.internal_sql.components import InternalManifestRow, INTERNAL_MANIFEST_COLUMNS, \
+    InternalManifestColumn
+from spinta.manifests.tabular.components import ManifestRow, MANIFEST_COLUMNS
+from spinta.manifests.tabular.helpers import ENUMS_ORDER_BY, sort, MODELS_ORDER_BY, DATASETS_ORDER_BY, \
+    to_relative_model_name, PROPERTIES_ORDER_BY, _get_type_repr, _read_tabular_manifest_rows
+from sqlalchemy_utils import UUIDType
+
+from spinta.spyna import unparse
+from spinta.types.datatype import Ref
+from spinta.utils.data import take
+from spinta.utils.schema import NotAvailable, NA
+from spinta.utils.types import is_str_uuid
+
+
+def read_schema(path: str):
+    engine = sa.create_engine(path)
+    with engine.connect() as conn:
+        yield from _read_all_sql_manifest_rows(path, conn)
+
+
+def get_table_structure(meta: sa.MetaData):
+    table = sa.Table(
+        '_manifest',
+        meta,
+        sa.Column("id", UUIDType, primary_key=True),
+        sa.Column("parent", UUIDType),
+        sa.Column("depth", sa.Integer),
+        sa.Column("path", sa.String),
+        sa.Column("mpath", sa.String),
+        sa.Column("dim", sa.String),
+        sa.Column("name", sa.String),
+        sa.Column("type", sa.String),
+        sa.Column("ref", sa.String),
+        sa.Column("source", sa.String),
+        sa.Column("prepare", sa.JSON),
+        sa.Column("level", sa.Integer),
+        sa.Column("access", sa.String),
+        sa.Column("uri", sa.String),
+        sa.Column("title", sa.String),
+        sa.Column("description", sa.String)
+    )
+    return table
+
+
+def _read_all_sql_manifest_rows(
+    path: Optional[str],
+    conn: sa.engine.Connection,
+    *,
+    rename_duplicates: bool = True
+):
+    meta = sa.MetaData(conn)
+    table = get_table_structure(meta)
+    stmt = sa.select([
+        table,
+        sa.literal_column("prepare IS NULL").label("prepare_is_null")]
+    )
+    rows = conn.execute(stmt)
+    converted = convert_sql_to_tabular_rows(list(rows))
+    yield from _read_tabular_manifest_rows(path=path, rows=converted, rename_duplicates=rename_duplicates)
+
+
+def write_internal_sql_manifest(dsn: str, manifest: Manifest):
+    engine = sa.create_engine(dsn)
+    with engine.connect() as conn:
+        meta = sa.MetaData(conn)
+        meta.reflect()
```

Comment:
Do not reflect whole database, use [Inspector.has_table()](https://docs.sqlalchemy.org/en/20/core/reflection.html#sqlalchemy.engine.reflection.Inspector.has_table).